Mathlib/Data/List/BigOperators/Basic.lean

```diff
@@ -27,50 +27,60 @@ section Monoid
 
 variable [Monoid M] [Monoid N] [Monoid P] {l l₁ l₂ : List M} {a : M}
 
-@[simp, to_additive]
-theorem prod_nil : ([] : List M).Prod = 1 :=
+@[to_additive (attr := simp)]
+theorem prod_nil : ([] : List M).prod = 1 :=
   rfl
 #align list.prod_nil List.prod_nil
 
 @[to_additive]
-theorem prod_singleton : [a].Prod = a :=
+theorem prod_singleton : [a].prod = a :=
   one_mul a
 #align list.prod_singleton List.prod_singleton
 
-@[simp, to_additive]
-theorem prod_cons : (a :: l).Prod = a * l.Prod :=
+@[to_additive (attr := simp)]
+theorem prod_cons : (a :: l).prod = a * l.prod :=
   calc
-    (a :: l).Prod = foldl (· * ·) (a * 1) l := by
-      simp only [List.prod, foldl_cons, one_mul, mul_one]
+    (a :: l).prod = foldl (· * ·) (a * 1) l :=
+      by simp only [List.prod, foldl_cons, one_mul, mul_one]
     _ = _ := foldl_assoc
-    
+
 #align list.prod_cons List.prod_cons
 
-@[simp, to_additive]
-theorem prod_append : (l₁ ++ l₂).Prod = l₁.Prod * l₂.Prod :=
+@[to_additive (attr := simp)]
+theorem prod_append : (l₁ ++ l₂).prod = l₁.prod * l₂.prod :=
   calc
-    (l₁ ++ l₂).Prod = foldl (· * ·) (foldl (· * ·) 1 l₁ * 1) l₂ := by simp [List.prod]
-    _ = l₁.Prod * l₂.Prod := foldl_assoc
-    
+    (l₁ ++ l₂).prod = foldl (· * ·) (foldl (· * ·) 1 l₁ * 1) l₂ := by simp [List.prod]
+    _ = l₁.prod * l₂.prod := foldl_assoc
+
 #align list.prod_append List.prod_append
 
 @[to_additive]
-theorem prod_concat : (l.concat a).Prod = l.Prod * a := by
+theorem prod_concat : (l.concat a).prod = l.prod * a := by
   rw [concat_eq_append, prod_append, prod_singleton]
 #align list.prod_concat List.prod_concat
 
-@[simp, to_additive]
-theorem prod_join {l : List (List M)} : l.join.Prod = (l.map List.prod).Prod := by
+@[to_additive (attr := simp)]
+theorem prod_join {l : List (List M)} : l.join.prod = (l.map List.prod).prod := by
   induction l <;> [rfl, simp only [*, List.join, map, prod_append, prod_cons]]
 #align list.prod_join List.prod_join
 
 @[to_additive]
-theorem prod_eq_foldr : l.Prod = foldr (· * ·) 1 l :=
-  (List.recOn l rfl) fun a l ihl => by rw [prod_cons, foldr_cons, ihl]
+theorem prod_eq_foldr : ∀ {l : List M}, l.prod = foldr (· * ·) 1 l
+  | [] => rfl
+  | cons a l => by rw [prod_cons, foldr_cons, prod_eq_foldr]
 #align list.prod_eq_foldr List.prod_eq_foldr
 
-@[simp, to_additive]
-theorem prod_repeat (a : M) (n : ℕ) : (repeat a n).Prod = a ^ n :=
+@[to_additive (attr := simp)]
+theorem prod_replicate (a : M) (n : ℕ) : (List.replicate n a).prod = a ^ n :=
```

Comment:
This seems not to be in the mathport output -- intentionally added?  Is `replicate` newly added, some variant of `repeat`?

Reply:
https://leanprover-community.github.io/mathlib4_docs/Init/Data/List/Basic.html#List.replicate